### PR TITLE
add release note for 09/2024 release

### PR DESCRIPTION
**Pull request by lukasvogel**

### CODE REVIEW

File: data/releasenotes.json [hunk 1]
```diff
@@ -1,4 +1,29 @@
 [
+	{
+		"version": "alpha/2024-09-13",
+		"date": "2024-09-13",
+		"breaksdb": true,
+		"notes": [
+			"<b>New Feature:</b> Implemented the functionality of the <a href=https://github.com/pgvector/pgvector>pg_vector</a> extension. For a feature description take a look at the <a href=../references/datatypes/vector>vector datatype reference</a>, the <a href=../references/advanced/pgvector>pg_vector extension compatibility docs</a>, or follow <a href=../example_datasets/glove>our new vector tutorial</a>.",
```
Comment (MoritzS) [inline data/releasenotes.json]:
```suggestion
			"<b>New Feature:</b> Implemented the functionality of the <a href=https://github.com/pgvector/pgvector>pgvector</a> extension. For a feature description take a look at the <a href=../references/datatypes/vector>vector datatype reference</a>, the <a href=../references/advanced/pgvector>pgvector extension compatibility docs</a>, or follow <a href=../example_datasets/glove>our new vector tutorial</a>.",
```

File: data/releasenotes.json [hunk 2]
```diff
@@ -1,4 +1,29 @@
...
+			"<b>New Feature:</b> Added support for quantified expressions over subqueries and arrays: <br /> <tt>with tmp(x) as (select array[1,2,3]) select 1 = ANY(x) from tmp</tt>.",
+			"<b>New Feature:</b> Implicitly cast string constants in binary expressions: You can now, for example, write <tt> 123 + '456'</tt>.",
+			"<b>New Feature:</b> Added support for row results in subqueries and non constant arrays: <br /> <tt>select (r.r).\"2\", (r.r).\"1\", (r.r).\"0\" from (select row('foo','bar','baz')) r(r)\"</tt> returns <tt> baz bar foo </tt>.",
+			"<b>Maintenance:</b> Introduce recovery log versioning to reduce the likelihood of having to break backwards compatibility in the future.",
+			"<b>Performance:</b> Implemented a heuristic that inlines CTEs that materialize a large state but are cheap to recalculate. This improves TPC-DS Q95's performance by about 10x, for example.",
+			"<b>Performance:</b> More aggressively use band joins where beneficial, even if the query does not contain <tt> between </tt> conditions.",
```
Comment (MoritzS) [inline data/releasenotes.json]:
```suggestion
			"<b>Performance:</b> More aggressively use band joins where beneficial, even if the query does not contain <tt>between</tt> conditions.",
```

**Review by MoritzS — APPROVED** [review]

File: data/releasenotes.json [hunk 3]
```diff
@@ -1,4 +1,29 @@
 [
+	{
+		"version": "alpha/2024-09-13",
+		"date": "2024-09-13",
+		"breaksdb": true,
+		"notes": [
+			"<b>New Feature:</b> Implemented the functionality of the <a href=https://github.com/pgvector/pgvector>pg_vector</a> extension. For a feature description take a look at the <a href=../references/datatypes/vector>vector datatype reference</a>, the <a href=../references/advanced/pgvector>pg_vector extension compatibility docs</a>, or follow <a href=../example_datasets/glove>our new vector tutorial</a>.",
+			"<b>New Feature:</b> Restructured database file storage: All files of a given database are stored in the same directory to declutter your server. Existing database files will be converted to the new format automatically on startup, no manual intervention required.",
+			"<b>New Feature:</b> Added support for quantified expressions over subqueries and arrays: <br /> <tt>with tmp(x) as (select array[1,2,3]) select 1 = ANY(x) from tmp</tt>.",
+			"<b>New Feature:</b> Implicitly cast string constants in binary expressions: You can now, for example, write <tt> 123 + '456'</tt>.",
```
Comment (pfent) [inline data/releasenotes.json:10]:
```suggestion
			"<b>New Feature:</b> Implicitly cast string constants in binary expressions: You can now, for example, write <tt> 123 + '456'</tt>.",
			"<b>New Feature:</b> Better type deduction for unknown types: Parameters now seamlessly work in function calls. For example <tt>lower($1)</tt> now treats the parameter as a string.",
```

File: data/releasenotes.json [hunk 4]
```diff
@@ -1,4 +1,29 @@
...
+			"<b>New Feature:</b> Added support for row results in subqueries and non constant arrays: <br /> <tt>select (r.r).\"2\", (r.r).\"1\", (r.r).\"0\" from (select row('foo','bar','baz')) r(r)\"</tt> returns <tt> baz bar foo </tt>.",
+			"<b>Maintenance:</b> Introduce recovery log versioning to reduce the likelihood of having to break backwards compatibility in the future.",
+			"<b>Performance:</b> Implemented a heuristic that inlines CTEs that materialize a large state but are cheap to recalculate. This improves TPC-DS Q95's performance by about 10x, for example.",
+			"<b>Performance:</b> More aggressively use band joins where beneficial, even if the query does not contain <tt> between </tt> conditions.",
+			"<b>Performance:</b> Optimize concat calls with single arguments: <tt> concat(x)</tt> becomes <tt> cast(x as text)</tt>.",
```
Comment (pfent) [inline data/releasenotes.json:17]: Optimizer: Improved cardinality estimation for very selective predicates. This sometimes improves execution plans, where we observed cases of 10x speedup

**Pull request merged**